src/main/scala/mesosphere/marathon/state/PathId.scala

```diff
@@ -101,41 +102,40 @@ object PathId {
   private[this] val ID_PATH_SEGMENT_PATTERN =
     "^(([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])\\.)*([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])|(\\.|\\.\\.)$".r
 
+  private val validChars = new Validator[PathId] {
+    override def apply(pathId: PathId): Result = {
+      validate(pathId.path)(validator = pathId.path.each should matchRegexFully(ID_PATH_SEGMENT_PATTERN.pattern))
+    }
+  }
+
   /**
     * For external usage. Needed to overwrite the whole description, e.g. id.path -> id.
     * @return
     */
-  implicit def pathIdValidator: Validator[PathId] = {
-    new Validator[PathId] {
-      override def apply(pathId: PathId): Result = {
-        validate(pathId.path)(validator = pathId.path.each should matchRegexFully(ID_PATH_SEGMENT_PATTERN.pattern)) and
-          validChild(pathId)
-      }
-    }
-  }
+  implicit val pathIdValidator = validator[PathId] { path =>
+    path is pathParentChildValidator(path.parent)
+  } and validChars
```

Comment:
Just curious: would that be the same as `path has validChars` inside the validator?